### ISSUE TITLE
behindSwipedItemLayout disappears on reinitializing the adapter

### ISSUE THREAD

**Issue by jb-102** [ernestoyaquello/DragDropSwipeRecyclerview#45]
View Used:: DragDropSwipeRecyclerView
Adapter Used:: DragDropSwipeAdapter

XML

        `<com.ernestoyaquello.dragdropswiperecyclerview.DragDropSwipeRecyclerView`
            android:layout_width="match_parent"
            android:id="@+id/rv_orders"
            app:behind_swiped_item_custom_layout="@layout/behind_swiped_vertical_list"
            app:behind_swiped_item_custom_layout_secondary="@layout/behind_swiped_vertical_list_secondary"
            android:layout_height="match_parent" />`

Main Activity 

    `private fun setupRecyclerView(orders: List<OrderSingle>) {
        if (swipeRefresh.isRefreshing) {
            swipeRefresh.isRefreshing = false
        }
        if(currentPage == 1) {
            mainOrdersList = orders.toMutableList()
            layoutManager = LinearLayoutManager(this)
            rv_orders.layoutManager = layoutManager
            adapter = OrdersAdapter(this, this, mainOrdersList, mainOrdersList, status!!)
            rv_orders.adapter = adapter
            adapter.notifyDataSetChanged()
            rv_orders.orientation = DragDropSwipeRecyclerView.ListOrientation.VERTICAL_LIST_WITH_VERTICAL_DRAGGING
            rv_orders.behindSwipedItemLayoutId = R.layout.behind_swiped_vertical_list
            rv_orders.behindSwipedItemSecondaryLayoutId = R.layout.behind_swiped_vertical_list_secondary
            rv_orders.reduceItemAlphaOnSwiping = true
            rv_orders.scrollListener = onListScrollListener
            rv_orders.dragListener = onItemDragListener
            rv_orders.swipeListener = onItemSwipeListener
            rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.UP)
            rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.DOWN)
            rv_orders.isNestedScrollingEnabled = true
            rv_orders.setHasFixedSize(true)
            when (status) {
                "NEW" -> {
                    rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.RIGHT)
                }
                "PENDING" -> {
                    rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.LEFT)
                }
                else -> {
                    rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.LEFT)
                    rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.RIGHT)
                }
            }
        } else {
            mainOrdersList.addAll(orders)
            Log.e("More loading","SIZE " + mainOrdersList.size)
            Log.e("More loading","SIZE 2 " + orders.size)
            adapter.notifyDataSetChanged()
        }
    }`

Adapter 

`class OrdersAdapter(
    private var context: Context,
    private var listener: OnItemClickListener,
    private var data: List<OrderSingle>,
    private var filteredData: List<OrderSingle>,
    private var statusText: String
) : DragDropSwipeAdapter<OrderSingle, OrdersAdapter.MyViewHolder>(filteredData), Filterable {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.single_item_order, parent, false)
        return MyViewHolder(view)
    }

    override fun getItemCount(): Int {
        Log.e("COUNT", filteredData.size.toString())
        return filteredData.size
    }

    class MyViewHolder(view: View) : DragDropSwipeAdapter.ViewHolder(view) {
        val singleOrderItem: LinearLayout = view.findViewById(R.id.singleOrderItem)
        val orderId: TextView = view.findViewById(R.id.tv_orderId_SO)
        val status: TextView = view.findViewById(R.id.tv_status_SO)
        val name: TextView = view.findViewById(R.id.tv_name_SO)
        private val address: TextView = view.findViewById(R.id.tv_address_SO)
        val dragIcon: ImageView = view.findViewById(R.id.drag_icon)

        fun bindTo(item: OrderSingle, context: Context, statusText: String) {
            try {
                orderId.text = "ORDER ID: ${item.orderId}"

                status.text = item.orderStatus
                if(item.orderStatus == "PICKUP") {
                    status.text = "REMAINING"
                }
                when (statusText) {
                    "PENDING", "NEW" -> {
                        status.setTextColor(ContextCompat.getColor(context, R.color.pending))
                    }
                    "DELIVERED" -> {
                        status.setTextColor(ContextCompat.getColor(context, R.color.delivered))
                    }
                    else -> {
                        status.setTextColor(ContextCompat.getColor(context, R.color.colorPrimary))
                    }
                }
                name.text = item.shipTo
                address.text = item.shippingAddress
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }

    override fun getFilter(): Filter {
        return object : Filter() {
            override fun performFiltering(charSequence: CharSequence): FilterResults? {
                val charString = charSequence.toString()
                if (charString.isEmpty()) {
                    filteredData = data
                } else {
                    val filteredList: MutableList<OrderSingle> = ArrayList()
                    for (row in data) { // name match condition. this might differ depending on your requirement
                        if (
                            row.orderId.toLowerCase(Locale.ROOT).contains(charString.toLowerCase(Locale.ROOT)) ||
                            row.shipTo.toLowerCase(Locale.ROOT).contains(charString.toLowerCase(Locale.ROOT)) ||
                            row.shippingAddress.toLowerCase(Locale.ROOT).contains(charString.toLowerCase(Locale.ROOT)) ||
                            row.orderStatus.toLowerCase(Locale.ROOT).contains(charString.toLowerCase(Locale.ROOT)) ||
                            row.deliveryType.toLowerCase(Locale.ROOT).contains(charString.toLowerCase(Locale.ROOT))
                                ) {
                            filteredList.add(row)
                        }
                    }
                    filteredData = filteredList
                }
                val filterResults = FilterResults()
                filterResults.values = filteredData
                return filterResults
            }

            override fun publishResults(
                charSequence: CharSequence?,
                filterResults: FilterResults
            ) {
                filteredData = filterResults.values as List<OrderSingle>
                notifyDataSetChanged()
            }
        }
    }

    override fun getViewHolder(itemView: View): MyViewHolder {
        return MyViewHolder(itemView)
    }

    override fun getViewToTouchToStartDraggingItem(
        item: OrderSingle,
        viewHolder: MyViewHolder,
        position: Int
    ) = viewHolder.dragIcon

    override fun onBindViewHolder(item: OrderSingle, viewHolder: MyViewHolder, position: Int) {
        viewHolder.bindTo(item, context, statusText)
        viewHolder.singleOrderItem.setOnClickListener {
            if(statusText != "NEW")
                listener.onItemClick(filteredData[position].orderId)
        }
    }

}`

Whenever the list pull pulled from top to bottom, new data set is called and that in turn calls 'setupRecyclerView' and provides it with new data list and as soon as I refresh the list, both the behindSwipedItemLayout disappears and I get empty background when swiped.

**Comment by jb-102** [comment]
`private fun setupRecyclerView(orders: List<OrderSingle>) {
        if (swipeRefresh.isRefreshing) {
            swipeRefresh.isRefreshing = false
        }
        if(currentPage == 1) {
            if(::mainOrdersList.isInitialized) {
                mainOrdersList.addAll(orders)
                adapter.notifyDataSetChanged()
                return
            }
            mainOrdersList = orders.toMutableList()
            layoutManager = LinearLayoutManager(this)
            rv_orders.layoutManager = layoutManager
            adapter = OrdersAdapter(this, this, mainOrdersList, mainOrdersList, status!!)
            rv_orders.adapter = adapter
            adapter.notifyDataSetChanged()
            rv_orders.orientation = DragDropSwipeRecyclerView.ListOrientation.VERTICAL_LIST_WITH_VERTICAL_DRAGGING
            rv_orders.behindSwipedItemLayoutId = R.layout.behind_swiped_vertical_list
            rv_orders.behindSwipedItemSecondaryLayoutId = R.layout.behind_swiped_vertical_list_secondary
            rv_orders.reduceItemAlphaOnSwiping = true
            rv_orders.scrollListener = onListScrollListener
            rv_orders.dragListener = onItemDragListener
            rv_orders.swipeListener = onItemSwipeListener
            rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.UP)
            rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.DOWN)
            rv_orders.isNestedScrollingEnabled = true
            rv_orders.setHasFixedSize(true)
            when (status) {
                "NEW" -> {
                    rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.RIGHT)
                }
                "PENDING" -> {
                    rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.LEFT)
                }
                else -> {
                    rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.LEFT)
                    rv_orders.disableSwipeDirection(DragDropSwipeRecyclerView.ListOrientation.DirectionFlag.RIGHT)
                }
            }
        } else {
            mainOrdersList.addAll(orders)
            Log.e("More loading","SIZE " + mainOrdersList.size)
            Log.e("More loading","SIZE 2 " + orders.size)
            adapter.notifyDataSetChanged()
        }
    }`

When swiped down for refresh, Just cleared the list(mainOrdersList) before hitting the API and then once I get the results, added the below code to the recycler view setup function.

if(::mainOrdersList.isInitialized) {
 mainOrdersList.addAll(orders)
 adapter.notifyDataSetChanged()
 return
}

Is working for me now. if you have a better solution. please let me know

**Comment by ernestoyaquello** [comment]
I am not sure why setting up a new adapter or updating the adapter's dataset would cause any issues, so there might be something in your implementation that is somehow causing this problem to happen. Unfortunately, there is a lot of code here, and it isn't even formatted properly, so I find it quite hard to follow. If you really cannot simplify it, maybe uploading a reproduction project would help.

**Comment by ernestoyaquello** [comment]
Given the lack of activity and the impossibility of reproducing the issue, I will close it.
